### PR TITLE
Make app channel address configurable instead of hard coded

### PR DESCRIPTION
**Pull request by yaron2**
Closes https://github.com/dapr/dapr/issues/6177.

### CODE REVIEW

File: pkg/actors/actors.go [hunk 1]
```diff
@@ -598,7 +598,7 @@ func (a *actorsRuntime) callRemoteActor(
 }
 
 func (a *actorsRuntime) isActorLocal(targetActorAddress, hostAddress string, grpcPort int) bool {
-	return strings.Contains(targetActorAddress, "localhost") || strings.Contains(targetActorAddress, "127.0.0.1") ||
+	return strings.Contains(targetActorAddress, "localhost") || strings.Contains(targetActorAddress, a.config.AppChannelAddress) ||
```
Comment (ItalyPaleAle) [inline pkg/actors/actors.go:601]: Any reason why localhost is hardcoded but 127.0.0.1 may not be present? If app channel isn't on localhost shouldn't that be missing too

**Review by ItalyPaleAle — CHANGES_REQUESTED** [review]
Please add an E2E test for this. Modifying an existing test (app health could be a good one since it uses a bunch of features of the app channel) may be enough

File: pkg/actors/actors.go (hunk 1 above)
Comment (yaron2) [reply pkg/actors/actors.go]: Further investigation has shown this section checks the `daprd` address and not the app channel, so this was unneeded. Changed it back

**Comment by mukundansundar** [comment]
@yaron2 this requires a change in CLI and the run template also ...

File: pkg/actors/actors.go (hunk 1 above)
Comment (yaron2) [reply pkg/actors/actors.go]: Resolving this now as the code has been reverted

**Comment by yaron2** [comment]
As FYI, our e2e test infrastructure didn't really support apps with the sidecar disabled, so fixed that along the way.

File: pkg/injector/annotations/annotations.go [hunk 2]
```diff
@@ -67,6 +67,7 @@ const (
 	KeyPluggableComponentsSocketsFolder = "dapr.io/pluggable-components-sockets-folder"
 	KeyPluggableComponentContainer      = "dapr.io/component-container"
 	KeyPluggableComponentsInjection     = "dapr.io/inject-pluggable-components"
+	KeyAppChannel                       = "dapr.io/app-channel-address"
```
Comment (artursouza) [inline pkg/injector/annotations/annotations.go:70]: Does this have any real life application? If this is added, then the app has a Deployment in K8s to be talked via localhost, not?

Comment (yaron2) [reply pkg/injector/annotations/annotations.go:70]: Yes, this tells Dapr to send events and messages to this configured address which is not localhost

**Review by daixiang0 — APPROVED** [review]

**Pull request merged**